cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/TaskWorkerServiceLauncher.java

```diff
@@ -195,6 +195,32 @@ public void run() {
     this.twillController = activeController;
   }
 
+  private SecurityContext createSecurityContext() {
+    SecurityContext.SecurityContextBuilder builder = new SecurityContext.SecurityContextBuilder();
+    String twillUserIdentity = cConf.get(Constants.Twill.Security.IDENTITY_USER);
+    if (twillUserIdentity != null) {
+      builder.withIdentity(twillUserIdentity);
+    }
+
+    Optional<Long> userId = Optional.empty();
+    try {
+      userId = Optional.of(cConf.getLong(Constants.TaskWorker.CONTAINER_RUN_AS_USER));
+    } catch (NullPointerException e) {
+      //no-op if configuration property does not exist
+    }
+    userId.ifPresent(id -> builder.withUserId(id));
+
+    Optional<Long> groupId = Optional.empty();
+    try {
+      groupId = Optional.of(cConf.getLong(Constants.TaskWorker.CONTAINER_RUN_AS_GROUP));
```

Comment:
probably we can assume if NPE not thrown here, then we can add groupId to builder? same for userId